server/app/services/rpc_client.rb

```diff
@@ -3,18 +3,22 @@
 class RpcClient
 
   class Error < StandardError
-    attr_accessor :code, :message, :backtrace
+    attr_reader :code, :remote_backtrace
 
-    def initialize(code, message, backtrace = nil)
-      self.code = code
-      self.message = message
-      self.backtrace = backtrace
+    def initialize(code, message, remote_backtrace = nil)
+      @code = code
+      @remote_backtrace = (['<Remote>:'] + remote_backtrace) if remote_backtrace
+      @caller = caller
+      super(message)
     end
-  end
 
-  class TimeoutError < Error
+    def backtrace
+      (super || @caller) + Array(remote_backtrace)
```

Comment:
All of the `raise Exception.new("foofoo")` which will otherwise get a nil backtrace.